Libraries/LibWeb/Internals/InternalsBase.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2025, Tim Flynn <trflynn89@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibWeb/Bindings/PlatformObject.h>
+#include <LibWeb/Forward.h>
+
+namespace Web::Internals {
+
+class InternalsBase : public Bindings::PlatformObject {
+    WEB_PLATFORM_OBJECT(InternalsBase, Bindings::PlatformObject);
+    GC_DECLARE_ALLOCATOR(InternalsBase);
+
+public:
+    virtual ~InternalsBase() override;
+
+protected:
+    explicit InternalsBase(JS::Realm&);
+
+    HTML::Window& window() const;
+    Page& page() const;
```

Comment:
Long term, I think we should probably have this getter on `PlatformObject`, one such FIXME being https://github.com/LadybirdBrowser/ladybird/blob/61c1e4a85519660b4784d4d5976d4e5d30410490/Libraries/LibWeb/WebSockets/WebSocket.cpp#L203

Only reason why I mention that is because `Processes` does not use `window()` either which makes me consider if we would need a base class if we did put `page()` in `PlatformObject`

Reply:
That probably makes sense long term, but like that comment alludes to, we don't have a way to get at a `Page` from the worker (at least as far as I can tell). Whereas at least in this current base class, we can assume we have a `Window`.

Reply:
Right, understood, my comment was more me thinking about the utility of having an entirely new `InternalsBase` for this considering it only really provides that function and a lot of places already in our code have just inlined that implementation (i.e put in `Processes`) to get the page.

Anyway, super minor point, very nice for us to move this approach as a special about page.